graphql_compiler/tests/test_ir_generation_errors.py

```diff
@@ -469,6 +469,18 @@ def test_recurse_directive_constraints(self):
             }
         }''')
 
+        # Note that ImportantEvent is a union of Event | BirthEvent, hence this query attempts
```

Comment:
nit: `ImportantEvent` sounds like a type, but you're actually referring to the edge `out_Animal_ImportantEvent`. This can introduce unnecessary confusion, so let's avoid abbreviations as much as possible.